omniport/core/open_auth/views/retrieve_data.py

```diff
@@ -0,0 +1,93 @@
+from rest_framework import status, generics
+from rest_framework.response import Response
+
+from oauth2_provider.models import AccessToken
+from kernel.models import Person
+
+from open_auth.models import Application
+from open_auth.utils import get_field_data, get_roles
+
+MODEL_REGEX = ['person', 'student', 'faculty_member', 'biological_information', 'contact_information']
+MODEL_STRINGS = ['person', 'person.student', 'person.facultymember', 'person.biologicalinformation', 'person.contact_information.first()']
+
+class GetUserData(generics.GenericAPIView):
+    """
+    View to retrieve data for oauth
+    based applications.
+    """
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Why is retrieval being handled in POST?

Reply:
We're asking for client id (and I've added client secret) and access token, so wouldn't it be better to handle these using POST rather than GET?

Reply:
It's still a request to fetch a resource. So an authenticated GET is the way to go.